src/devices/lacrosse_tx34.c

```diff
@@ -0,0 +1,158 @@
+- Modulation: FSK/PCM
+- Bit duration: 58 µs
+- Frame size: 64 bits (including preamble)
+
+Frame format:
+
+-------------
+| 1010 1010 | preamble (up to the first two bits may be lost)
+-------------
+| 0010 1101 | 0x2DD4: LaCrosse IT frame identifier
+| 1101 0100 |
+-------------
+| MMMM DDDD | MMMM: sensor model (5 for rain gauge, 9 for thermo/hydro...)
+| DDNW 0000 | DDDDDD: device ID (0 to 63, random at startup)
+| GGGG GGGG | N: new battery (on for about 420 minutes after startup)
+| GGGG GGGG | W: weak battery (on when battery voltage < 2 volts)
+------------- GGGGGGGGGGGGGGGG: bucket tipping counter
+| CCCC CCCC | CCCCCCCC: CRC8 on previous 4 bytes
+-------------
+
+This decoder decodes generic LaCrosse IT+ frames and filters TX34 ones.
+Could be merged with existing TX29 decoder... or not.
+*/
+
+#include "decoder.h"
+
+#define LACROSSE_TX34_CRC_POLY 0x31
+#define LACROSSE_TX34_CRC_INIT 0x00
+#define LACROSSE_TX34_PREAMBLE_BITS 22
+#define LACROSSE_TX34_ITMODEL 5
+#define LACROSSE_TX34_PAYLOAD_BITS 40
+#define LACROSSE_TX34_RAIN_FACTOR 0.222
+
+static int lacrosse_tx34_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+    int row;
+    uint8_t payload[5];
+    uint8_t r_crc, c_crc;
+    uint8_t sensor_id, new_bat, weak_bat;
+    uint16_t rain_tick;
+    float rain_mm;
+    int events;
+
+    // 22 bits preamble (shifted left): 101010b 0x2DD4
+    static const uint8_t preamble[] = { 0xA8, 0xB7, 0x50 };
+
+    // process rows
+    events = 0;
+    for (row = 0; row <= bitbuffer->num_rows; ++row)
+    {
+        unsigned int start_pos, payload_bits;
+
+        // search for preamble
+        start_pos = bitbuffer_search(bitbuffer, row, 0, preamble,
+            LACROSSE_TX34_PREAMBLE_BITS);
+        if (start_pos == bitbuffer->bits_per_row[row])
+        continue; // preamble not found
+        payload_bits = bitbuffer->bits_per_row[row] - start_pos -
+        LACROSSE_TX34_PREAMBLE_BITS;
+        if (payload_bits < LACROSSE_TX34_PAYLOAD_BITS)
+        continue; // probably truncated frame
+        if (decoder->verbose)
+        fprintf(stderr,
+            "LaCrosse IT frame detected (%d bits payload)\n",
+            payload_bits);
+
+        // get payload
+        bitbuffer_extract_bytes(bitbuffer, row,
+            start_pos + LACROSSE_TX34_PREAMBLE_BITS,
+            payload, LACROSSE_TX34_PAYLOAD_BITS);
+        // verify CRC
+        r_crc = payload[4];
+        c_crc = crc8(&payload[0], 4,
+            LACROSSE_TX34_CRC_POLY, LACROSSE_TX34_CRC_INIT);
+        if (r_crc != c_crc)
+        {
+            // bad CRC: reject IT frame
+            fprintf(stderr,
+                "LaCrosse IT frame bad CRC: calculated %02x, "
+                "received %02x\n",
+                c_crc, r_crc);
+            continue;
+        }
+
+        // check model
+        if (((payload[0] & 0xF0) >> 4) != LACROSSE_TX34_ITMODEL)
+            continue; // not a rain gauge...
+        if (decoder->verbose)
```

Comment:
remove this

Reply:
I wont. It would accept frame of every kind of IT+ sensor (thermo/hydro, tea pot...) instead of rain gauges only.

Reply:
I mean this `if` with `fprintf` below ;)

Reply:
Please could you give a reason for this removal? There is same debug print in other decoders (`like lacrosse_tx35.x`).

Reply:
Well, the next thing we do here is `decoder_output_data()` so `model: LaCrosse TX34-IT` will already be on the screen.
Yes, other decoders need a cleanup for fewer debug prints someday.

Reply:
OK